src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -548,6 +548,15 @@ private void ParseHelper(string[] args)
                     _noInteractive = true;
                     _skipUserInit = true;
                     _noExit = false;
+
+                    ++i;
+                    if (i < args.Length)
+                    {
+                        if (LanguagePrimitives.TryConvertTo<int>(args[i], out int verNumber) && verNumber == 2)
```

Comment:
I think we should currently error if any value is passed to `-v`

Reply:
Now we write the error and then show current version.
Please clarify your request.

Reply:
Aren't you only writing error if `-v 2`?  Should also error if `-v 3`

Reply:
I agree. Will add the error for `-v 3` too.

Reply:
We should error on any value passed to `-v`.  In some future time if we decide to support launching different installed versions of PSCore6, we can decide if we want to overload `-v` or even introduce a new parameter.

Reply:
Currently we ignore all values after `-v`. The docs says the same: accept only 2.0 or 3.0. If anybody use the parameter the breaking change is related only to 2.0 or 3.0 values - we should write an error for the cases only.

Reply:
Could you comment #4836 - I am ready to implement enumeration of installed versioms.

Reply:
If we currently ignore all values after -v, then it seems logical to write error on any value passed to -v saying that parameter doesn't have any effect.
Having a parameter that quietly does nothing doesn't look good.

Reply:
It is an enhancement. It make sense. Thanks!

Done.